conda/lock.py

```diff
@@ -33,10 +33,12 @@ class Locked(object):
     """
     Context manager to handle locks.
     """
-    def __init__(self, path, retries=10):
+    def __init__(self, path, file_tmp, retries=10):
```

Comment:
I think we discussed this before, but remind me... Why is it necessary to add the extra `file_tmp` argument here?

I'd propose that we leave the original `Locked` class as is so that we don't break its use in other parts of the code.  Then we take your version of `Locked` here you've modified it and call it `FileLock`.  We can give instructions in the documentation and assume that `path` is the location of a current or future file.  So, one entry into the context, conda creates a temporary lock file at the path `path + LOCKFN`, and deletes that lock file on exit.

I guess the original class added the PID to the lock file path name.  I like that actually, but only for informational purposes.  The behavior of the lock should *not* be influenced by PID--if a file path is locked, then it's locked, regardless of PID.  Otherwise that's kind of a dumb implementation of a lock.